### DISCUSSION TITLE
cefsharp DownloadUrlAsync skips Cookie

### DISCUSSION THREAD

**Discussion by moslem7026** [cefsharp/CefSharp#4578] [General]
load url in browser control that require cookie and authentiacaion is ok

`  CefSharp.OffScreen.ChromiumWebBrowser browser = new CefSharp.OffScreen.ChromiumWebBrowser(Url, requestContext: reqContext);`
but using download url

```
            var ppp =await browser.GetMainFrame().DownloadUrlAsync(Url);
            var ooop= Encoding.UTF8.GetString(ppp);
```
redirect to login page and cookie header not sent

**Comment by amaitland** [comment]
This is expected at the moment. It might be possible to add support for sending cookies.

Try adding the `UrlRequestFlags.AllowStoredCredentials` flag to the request in `DownloadUrlAsync`

Example https://github.com/cefsharp/CefSharp/blob/c09f10092d39be4f785b572a97f8e854aa78cf7c/CefSharp/WebBrowserExtensions.cs#L885

**Reply by amaitland** [reply]
A new optional param has been added to allow passing in `UrlRequestFlags` in version `116.0.150`. 

https://github.com/cefsharp/CefSharp/commit/a7088ebf9a011a2dad0ba5c7095ee1b7988c1e9e
